subprojects/logging/src/main/java/org/gradle/internal/logging/console/BuildStatusRenderer.java

```diff
@@ -119,10 +132,15 @@ private void phaseStarted(ProgressStartEvent progressStartEvent) {
         timerEnabled = true;
         currentPhaseProgressOperationId = progressStartEvent.getProgressOperationId().getId();
         currentBuildStatus = progressStartEvent.getShortDescription();
+        currentProgressIndicator = "";
     }
 
     private void phaseProgressed(ProgressEvent progressEvent) {
         currentBuildStatus = progressEvent.getStatus();
+        currentProgressIndicator = progressEvent.getProgressIndicator();
+        if (progressEvent.getBuildHealth() == BuildHealth.FAILING) {
+            currentBuildStatusColor = Style.Color.RED;
```

Comment:
`StyledLabel` needs to use `StyledTextOutputEvent.Span`. I can not see why it shouldn't. I changed it: https://github.com/gradle/gradle/pull/2447/commits/f9e9950f9e3c0a428e3ada721a7e2746bdabf057